osu.Game/Graphics/UserInterface/FPSCounter.cs

```diff
@@ -0,0 +1,234 @@
+                                },
+                            }
+                        },
+                        msCounter = new FrameTimeCounter
+                        {
+                            Anchor = Anchor.TopRight,
+                            Origin = Anchor.TopRight,
+                            Margin = new MarginPadding(1),
+                            Y = -1,
+                        },
+                        fpsCounter = new FramesPerSecondCounter
+                        {
+                            Anchor = Anchor.TopRight,
+                            Origin = Anchor.TopRight,
+                            Margin = new MarginPadding(1),
+                            Y = 10,
+                            Scale = new Vector2(0.8f),
+                        }
+                    }
+                },
+            };
+
+            config.BindWith(OsuSetting.ShowFpsDisplay, showFpsDisplay);
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            displayTemporarily();
+
+            showFpsDisplay.BindValueChanged(showFps =>
+            {
+                State.Value = showFps.NewValue ? Visibility.Visible : Visibility.Hidden;
+                if (showFps.NewValue)
+                    displayTemporarily();
+            }, true);
+
+            State.BindValueChanged(state => showFpsDisplay.Value = state.NewValue == Visibility.Visible);
+        }
+
+        protected override void PopIn() => this.FadeIn(100);
+
+        protected override void PopOut() => this.FadeOut(100);
+
+        protected override bool OnHover(HoverEvent e)
+        {
+            background.FadeTo(1, 200);
+            displayTemporarily();
+            return base.OnHover(e);
+        }
+
+        protected override void OnHoverLost(HoverLostEvent e)
+        {
+            background.FadeTo(idle_background_alpha, 200);
+            displayTemporarily();
+            base.OnHoverLost(e);
+        }
+
+        private bool isDisplayed;
+
+        private ScheduledDelegate? fadeOutDelegate;
+
+        private void displayTemporarily()
+        {
+            if (!isDisplayed)
+                mainContent.FadeTo(1, 300, Easing.OutQuint);
+
+            fadeOutDelegate?.Cancel();
+
+            if (!IsHovered)
+            {
+                fadeOutDelegate = Scheduler.AddDelayed(() =>
+                {
+                    mainContent.FadeTo(0, 300, Easing.OutQuint);
+                    isDisplayed = false;
+                }, 2000);
+            }
+        }
+
+        [Resolved]
+        private GameHost gameHost { get; set; } = null!;
+
+        protected override void Update()
+        {
+            base.Update();
+
+            // TODO: this is wrong (elapsed clock time, not actual run time).
+            double newFrameTime = gameHost.UpdateThread.Clock.ElapsedFrameTime;
+            double newFps = gameHost.DrawThread.Clock.FramesPerSecond;
```

Comment:
Asking about this from curiosity, but I would imagine this to be correct given that the elapsed frame time is computed from the time spent in one update frame? Can you explain further what the difference is between using elapsed clock time and actual run time?

Reply:
Ah so it's about using `Clock.ElapsedFrameTime` rather than the full time spent that's collected by `PerformanceMonitor`?

Reply:
Arguably you'd call the `PerformanceMonitor` the less "full" time.

Both are useful in some ways, so I'm not sure which to show to the user. `ElapsedFrameTime` is just `1/fps` though.